ui/src/components/CopyButton/CopyButton.js

```diff
@@ -18,11 +18,64 @@ export class CopyButton extends React.Component {
   }
 }
 
+export class CopyButton extends Component {
+  constructor(props) {
+    super(props);
+    this.INITIAL_STATE = {
+      confirmationPopupIsOpen: false,
+      confirmationPopupMsg: '',
+    };
+    this.state = this.INITIAL_STATE;
+    this.contextRef = React.createRef();
+  }
+
+  onCopy = () => {
+    this.setState(() => ({
+      confirmationPopupIsOpen: true,
+      confirmationPopupMsg: 'Copied!',
+    }));
+    this.delayClosePopup();
+  };
+
+  delayClosePopup = () => {
```

Comment:
could you tell what was the problem that is solved here?  just to be aware for the future

Reply:
True! So here how it works:
1. when you mouse over, it shows a popup saying "Copy to clipboard"
2. when you click, this popup disappears, and it shows new popup saying "Copied!"
3. this last one appears for a few seconds, then it disappears